File: C/caesar_cipher_encryption.c
#include <stdio.h>
#include <string.h>

void caesarCipher(char *text, int shift);

int main() {
    char plaintext[100];
    int shift;

    // Input plaintext
    printf("Enter plaintext: ");
    fgets(plaintext, sizeof(plaintext), stdin);

    // Remove trailing newline if present
    if (plaintext[strlen(plaintext) - 1] == '\n')
        plaintext[strlen(plaintext) - 1] = '\0';

    // Input shift amount
    printf("Enter shift amount (1-25): ");
    scanf("%d", &shift);

    // Encrypt the plaintext using Caesar cipher
    caesarCipher(plaintext, shift);

    printf("Cipher text: %s\n", plaintext);

    return 0;
}

void caesarCipher(char *text, int shift) {
    int i;
    for (i = 0; text[i] != '\0'; i++) {
        // Check if the character is a letter
        if (text[i] >= 'A' && text[i] <= 'Z') {
            // Encrypt uppercase letters
            text[i] = 'A' + (text[i] - 'A' + shift) % 26;
        } else if (text[i] >= 'a' && text[i] <= 'z') {
            // Encrypt lowercase letters
            text[i] = 'a' + (text[i] - 'a' + shift) % 26;
        }
    }
}

File: PYTHON/caesar_cipher_encryption.py
def caesar_cipher(text, shift):
    result = ""
    for char in text:
        if char.isalpha():  # Check if the character is a letter
            if char.isupper():  # Encrypt uppercase letters
                result += chr((ord(char) - ord('A') + shift) % 26 + ord('A'))
            else:  # Encrypt lowercase letters
                result += chr((ord(char) - ord('a') + shift) % 26 + ord('a'))
        else:
            result += char  # Keep non-letter characters unchanged
    return result

def main():
    plaintext = input("Enter plaintext: ")
    shift = int(input("Enter shift amount (1-25): "))

    # Encrypt the plaintext using Caesar cipher
    ciphertext = caesar_cipher(plaintext, shift)

    print("Cipher text:", ciphertext)

if __name__ == "__main__":
    main()
